browser/net/ipfs_redirect_network_delegate_helper.cc

```diff
@@ -42,4 +44,36 @@ int OnBeforeURLRequest_IPFSRedirectWork(
   return net::OK;
 }
 
+int OnHeadersReceived_IPFSRedirectWork(
+    const net::HttpResponseHeaders* response_headers,
+    scoped_refptr<net::HttpResponseHeaders>* override_response_headers,
+    GURL* allowed_unsafe_redirect_url,
+    const brave::ResponseCallback& next_callback,
+    std::shared_ptr<brave::BraveRequestInfo> ctx) {
+  if (!ctx->browser_context ||
+      IsIpfsResolveMethodDisabled(ctx->browser_context)) {
+    return net::OK;
+  }
+
+  std::string ipfs_path;
+  if (ctx->ipfs_auto_fallback && response_headers &&
```

Comment:
After @yrliou added browser_context to BraveRequestInfo [1] we should avoid blowing up `BraveRequestInfo` with new fields and just extract prefs from the context.

https://github.com/brave/brave-core/pull/7307/files#diff-5a208de77796edde3a97ada956a4cd34b8b55426786d2920c14931553653ed2dR18

Reply:
I think that PR happened at the same time as this one was already coded.  I can do a follow-up though.